wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/InternalEnvironmentFactory.java

```diff
@@ -106,11 +113,23 @@ public T create(final Environment sourceEnv) throws InfrastructureException, Val
 
     machinesValidator.validate(machines);
 
-    return doCreate(recipe, machines, warnings);
+    final T environment = doCreate(recipe, machines, warnings);
+
```

Comment:
Are you going to cover this class by unit tests?

Reply:
done